templates/activity_stream_derived/impression_stats_flat_v1/query.sql

```diff
@@ -0,0 +1,60 @@
+WITH base AS (
+  SELECT
+    *
+  FROM
+    activity_stream_stable.impression_stats_v1
+  WHERE
+    (@submission_date IS NULL OR @submission_date = DATE(submission_timestamp))
+),
+impression_data AS (
+  SELECT
+    *
+  FROM
+    base
+  WHERE
+    -- make sure data is valid/non-empty
+    ARRAY_LENGTH(tiles) >= 1
+)
```

Comment:
these CTEs seem unnecessary. the `ARRAY_LENGTH(tiles) >= 1` condition is redundant with `UNNEST(impression_data.tiles)` and the `submission_date` filter should work fine on the final query.